fluent-dom/src/localization.js

```diff
@@ -195,34 +195,28 @@ function valueFromBundle(bundle, errors, id, args) {
  * The function will return an object with a value and attributes of the
  * entity.
  *
- * If the function fails to retrieve the entity, the value is set to the ID of
- * an entity, and attributes to `null`. If formatting fails, it will return
- * a partially resolved value and attributes.
- *
- * In both cases, an error is being added to the errors array.
- *
  * @param   {FluentBundle} bundle
- * @param   {Array<Error>}   errors
- * @param   {String}         id
- * @param   {Object}         args
+ * @param   {Array<Error>} errors
+ * @param   {Object} message
+ * @param   {Object} args
  * @returns {Object}
  * @private
  */
-function messageFromBundle(bundle, errors, id, args) {
-  const msg = bundle.getMessage(id);
-
+function messageFromBundle(bundle, errors, message, args) {
   const formatted = {
-    value: bundle.format(msg, args, errors),
+    value: null,
     attributes: null,
   };
 
-  if (msg.attrs) {
+  if (message.value) {
+    formatted.value = bundle.formatPattern(message.value, args, errors);
+  }
+
+  if (message.attributes) {
     formatted.attributes = [];
```

Comment:
use `formatted.attributes = new Array(message.attributes.length());` to avoid reallocation.

Reply:
Oh, this actually made me realize that there's a bug here. `message.attributes` is a non-nullable object now, so the `if (message.attributes)` check always passes.

Reply:
Aaand fixed. Thanks!